Feat/crosschain savings

swiftcc624 opened the pull request.

parrot9design commented:
- [ ] Let's add a `$` to the Savings and Rewards values:
![Screenshot 2025-02-24 at 5 19 19 PM](https://github.com/user-attachments/assets/23865ed4-8f50-419d-a241-c85bd991e20c)

---

- [ ] Update the onboarding text for non-ICON chains:
![Screenshot 2025-02-24 at 4 54 15 PM](https://github.com/user-attachments/assets/e64c9321-3eb2-46cf-9824-563117cab098)

Earning potential:
>Switch to ICON, then lock up BALN to boost your rewards.

*For the `Switch to ICON` link, let's change to the ICON view without forcing the wallet connection.

Network fees:
>Lock up BALN on ICON to earn fees from Balanced transactions.


- [ ] Update the Rewards message in the signed out state:
- ICON: `To earn rewards on ICON: supply liquidity, deposit bnUSD into the Savings Rate, or lock up BALN.`
- Archway & Havah: No change needed
- Most chains: `To earn rewards on {chain}, supply liquidity or deposit bnUSD into the Savings Rate.`
![Screenshot 2025-02-24 at 4 58 39 PM](https://github.com/user-attachments/assets/73fd2925-bc1e-437d-a3f9-d5122fbd072a)![Screenshot 2025-02-24 at 4 59 16 PM](https://github.com/user-attachments/assets/cef1145b-0139-4c16-afad-36bbcad5cd3b)


---

- [ ] Remove the`...` from the confirmation button in the deposit/withdraw modals:

![Screenshot 2025-02-24 at 5 01 25 PM](https://github.com/user-attachments/assets/5a77c81c-f7c5-4e23-8b76-73fadc82c9b0)

---

- [ ] Halfway through a transaction (I think when it confirms on xcallscan, but I've been jumping between windows/apps during my review so that could also be to blame), the modal will update with the wrong information. 

For example, it went into the withdraw state with different values instead of completing the deposit transaction:
![Screenshot 2025-02-24 at 5 03 21 PM](https://github.com/user-attachments/assets/e107c173-e8a3-47d4-a704-64a42e6ea2d5)

Another time, the amount of bnUSD in the modal changed from 4.24 to 0.78 (my remaining balance), while the true amount was displayed as pending under Bridge > Activity. 


---

- [ ] Display the Savings Rate onboarding text when no bnUSD has been deposited:
![Screenshot 2025-02-24 at 4 52 27 PM](https://github.com/user-attachments/assets/dbd0f34c-8d95-42af-9d86-8f0e5740af10)


---

- [ ] Can we adjust the sizing of the dropdown on mobile to prevent the chains from spreading over 2 lines? In the second image, I dropped the column labels + chain names by 2px.
![Screenshot 2025-02-24 at 5 26 45 PM](https://github.com/user-attachments/assets/95fe0af6-9e24-417c-858c-6f2b100deeab) ![Screenshot 2025-02-24 at 5 26 26 PM](https://github.com/user-attachments/assets/044ab919-dbc5-4933-8105-e965411bb0f4)

swiftcc624 commented:
@parrot9design 

> Display the Savings Rate onboarding text when no bnUSD has been deposited:

can you please elaborate more?

>  Can we adjust the sizing of the dropdown on mobile to prevent the chains from spreading over 2 lines? In the second image, I dropped the column labels + chain names by 2px.

can you explain `In the second image, I dropped the column labels + chain names by 2px.`?

I've fixed other issues

parrot9design commented:
>>Display the Savings Rate onboarding text when no bnUSD has been deposited:
>
>can you please elaborate more?

If you haven't deposited any bnUSD into the Savings Rate, you should see the Savings Rate onboarding text in the panel underneath, not the placeholders for sICX, BALN, and bnUSD. You aren't earning any rewards, so the UI shouldn't act like they could be added to your balance at any moment.

>>Can we adjust the sizing of the dropdown on mobile to prevent the chains from spreading over 2 lines? In the second image, I dropped the column labels + chain names by 2px.
>
>can you explain `In the second image, I dropped the column labels + chain names by 2px.`?

The second image shows the dropdown with the column labels + chain names reduced by 2px so that the information could fit comfortably within the space, rather than having some chain names split over two lines. We should only use the smaller font size for this dropdown when the screen size is below a certain width.

swiftcc624 commented:
@parrot9design I've fixed all the issues

parrot9design commented:
- [ ] Can we show the amount available to deposit, like we do in the collateral dropdown?
![Screenshot 2025-02-26 at 4 41 02 PM](https://github.com/user-attachments/assets/665d1a60-6c7d-44dd-be61-92414fbab2b4)


- [ ] Let's tweak the dropdown sizing at mobile:
![Screenshot 2025-02-26 at 4 43 57 PM](https://github.com/user-attachments/assets/8d5f9653-b259-4b10-b126-b905961f13b2)

- [ ] I deposited bnUSD on Sui, which finalised in like 20 seconds, but it doesn't show my position in the UI:
![Screenshot 2025-02-26 at 5 19 17 PM](https://github.com/user-attachments/assets/a5735d38-b91a-4633-be74-6ec6a4faee5f)
![Screenshot 2025-02-26 at 5 20 47 PM](https://github.com/user-attachments/assets/a432322e-d394-4779-a8f9-41642ac0c81b)

Same thing happened with Stellar, so I assume it's related to the xcallscan implementation?

parrot9design commented:
- [ ] Until you click one of the column headings, it doesn't show you how the list is being sorted. Let's default to the Savings column.
<img width="350" alt="Screenshot 2025-02-27 at 9 25 39 AM" src="https://github.com/user-attachments/assets/800f9b93-fd94-456a-a92c-ad9b4c4cbb6f" />
<img width="350" alt="Screenshot 2025-02-27 at 9 36 19 AM" src="https://github.com/user-attachments/assets/f183fc27-56f4-45e0-b5b2-a9847f7caf8a" />

---

- [ ] My Sui and Stellar savings rate positions still don't appear in the UI. I don't know what happened to the bnUSD I deposited...

---

- [ ] Can we remove Archway and Havah from the Rewards section? BALN and sICX aren't compatible, so even if you deposit bnUSD on these chains, you're not going to earn any rewards for it.
<img width="1084" alt="Screenshot 2025-02-27 at 9 30 12 AM" src="https://github.com/user-attachments/assets/9ab7168e-7042-4982-833b-c98e065098f2" />
<img width="1057" alt="Screenshot 2025-02-27 at 9 35 02 AM" src="https://github.com/user-attachments/assets/6c781bbf-c7ec-421e-bbf2-8af3c36e4007" />

---

- [ ] There's a weird bug with the dropdown on mobile. When I first checked, `Avalanche` was spread across 2 lines, but when I moved from Hana to Safari and back again, the sizing changed so it was on one line. Switch between apps again, and it's back to 2 lines. Looks like it's an issue with the Rewards column alignment:

<img width="350" src="https://github.com/user-attachments/assets/7339ebb0-de58-48c5-acba-713fba4cd88e"><img width="350" src="https://github.com/user-attachments/assets/97d26f93-81e1-48b0-8787-b6b0e6141701">

parrot9design commented:
- [ ] After the first send transaction was complete, the second Send button didn’t become active. 
![Supply liquidity](https://github.com/user-attachments/assets/a27e5810-54fa-4727-a0d8-c3188f164085)

- [ ] When I then removed the XLM, the transaction finalised on xcallscan & the XLM was added to my wallet, but the modal remained stuck in a pending state. Same thing happened when trying to remove BTC & bnUSD on Arbitrum.
![Supply liquidity](https://github.com/user-attachments/assets/af4f7264-37cb-4449-84e8-416361d4372b)

- [ ] Many transactions remain pending in the UI long after a transaction is complete on xcallscan. For liquidity transactions, I had to refresh the page to close out of the modal, but the details remain in the Activity section:
![Arbitrum → ICON](https://github.com/user-attachments/assets/b2aaa051-5ffc-4b19-912c-aefe16e3c54f)

- [ ] How are my daily rewards higher than the total?
![0 2464 AVAX](https://github.com/user-attachments/assets/05740184-930d-48cc-bfb7-80b84e093d50)

- [ ] Change `Rewards` to `rewards` here:
![Screenshot 2025-03-05 at 1 26 49 PM](https://github.com/user-attachments/assets/6a85e18a-69eb-4936-84b2-3fd748eeaa3c)

- [ ] Change `Not now` to `Cancel` here:
![Claim Savings Rate rewards](https://github.com/user-attachments/assets/773a5eff-2dc3-428e-9323-da3c260a0f94)

- [ ] Still noticing a few issues with the dropdown on mobile. The chevron spacing is tight for the Savings & Rewards columns, and Avalanche alternates between 1 & 2 lines.
![Rewards](https://github.com/user-attachments/assets/17b11b61-1158-480e-879b-3ce8f8700657)

- [ ] If people aren’t doing anything to earn rewards on a specific chain, we should show the same onboarding text you see in the signed out state. Compare the first image to how it works when signed in on the live app:
![Screenshot 2025-03-05 at 1 35 26 PM](https://github.com/user-attachments/assets/8a346655-118f-4719-a784-04be99631647)
![Screenshot 2025-03-05 at 1 36 37 PM](https://github.com/user-attachments/assets/3e5d89e5-5d10-4df3-87d0-48f97fff194d)

parrot9design commented:
- [ ] These should be `Cancel` to be consistent with the Savings rate claim modal:
![Screenshot 2025-03-11 at 1 51 14 PM](https://github.com/user-attachments/assets/3eefd9c7-246b-4db5-a7d5-323dc27c7a14)![Screenshot 2025-03-11 at 1 51 37 PM](https://github.com/user-attachments/assets/d12a621a-41ce-4054-a9df-4137f6496c32)

- [ ] The `Cancel` button should change to `Close` _after_ the transaction has been signed, when it's no longer possible to cancel the transaction. It currently changes as soon as you click the Claim button. 

https://github.com/user-attachments/assets/d9b6c635-6c6b-429c-ac0b-db8348c61433

<br>

- [ ] The Rewards dropdown still needs a little bit of tweaking on mobile. The spacing is much better when the columns are sorted low > high, but it's still tight when set from high > low.

![Screenshot 2025-03-11 at 2 16 41 PM](https://github.com/user-attachments/assets/d009b0dd-c21d-413a-86ee-c094fff74681) ![Screenshot 2025-03-11 at 2 16 48 PM](https://github.com/user-attachments/assets/5f5f1d16-358a-41e5-b56e-5fe9443afd5e)
![Screenshot 2025-03-11 at 2 16 55 PM](https://github.com/user-attachments/assets/028b5e15-d6e1-4440-980b-72c7fd875619) ![Screenshot 2025-03-11 at 2 17 00 PM](https://github.com/user-attachments/assets/40b2b72e-4add-40c4-a31c-d17d41e320ed)

- [ ] On mobile, the Avalanche chain still starts with the `e` on a second line, and fixes itself when you make any adjustments.

https://github.com/user-attachments/assets/3411466e-def5-4194-b28c-31d69905e4e1

- [ ] On mobile, the `Liquidity rewards` section has a tooltip next to it:
![Screenshot 2025-03-11 at 2 17 27 PM](https://github.com/user-attachments/assets/cb2e0284-1932-4797-a8a1-28ef4e08beb6)


- [ ] After I withdrew liquidity for SUI/bnUSD, the page glitched and acted like I had no liquidity and there was no SUI/bnUSD pool:

![Screenshot 2025-03-11 at 2 52 11 PM](https://github.com/user-attachments/assets/8b59c479-f97b-42e0-bc0d-908a490b5b44)

After about 10 seconds, it updated to the correct values:
![Screenshot 2025-03-11 at 2 52 23 PM](https://github.com/user-attachments/assets/ee67bdc8-a1a3-4dbb-acef-7c27e527dab9)

- [ ] The Rewards section shows that liquidity rewards are pending on Stellar, but no rewards are available for XLM/sICX:
![Screenshot 2025-03-11 at 3 53 55 PM](https://github.com/user-attachments/assets/88d6e1a0-a8ff-4ea8-bf32-cdc522af5a99)
![Screenshot 2025-03-11 at 3 54 02 PM](https://github.com/user-attachments/assets/bde7e69c-4290-4a2a-8f6e-31028b005298)

It does not show as `Pending` in the dropdown, though (the correct behaviour):
![Screenshot 2025-03-11 at 3 55 33 PM](https://github.com/user-attachments/assets/c1d4b7e5-f40c-4ee7-9371-81efdc99fbbc)

parrot9design commented:
- [ ] The rewards chain dropdown should show that rewards are being earned for network fees on ICON:
![Screenshot 2025-03-12 at 3 26 00 PM](https://github.com/user-attachments/assets/4ded106d-08fc-49aa-81b2-286dfd3f9f5f)


- [ ] After I withdrew 40.32 bnUSD from the savings rate on Sui and the transaction complete in the app and on xcallscan, the UI updated to show that I now had _80.65_ bnUSD: 
![Screenshot 2025-03-12 at 4 08 51 PM](https://github.com/user-attachments/assets/7ea714a7-6f31-4623-9961-5902e35ece9d)

It remained that way for a while as I did a sanity check, knowing that I'd only had 40.32 bnUSD on Sui. Then I tried to withdraw the rest:
![Screenshot 2025-03-12 at 4 09 49 PM](https://github.com/user-attachments/assets/1c8f4ca9-d331-44a7-b342-ec9dce495169)

That transaction failed (see https://xcallscan.xyz/messages/43232), but it shouldn't have been possible to make it in the first place.

- [ ] Update the Activity messages for depositing/withdrawing bnUSD. They should say `Deposit bnUSD` and `Withdraw bnUSD`.

- [ ] Noting in case it's something we can fix now or in a future release: the delay between the liquidity withdrawal complete message and the assets appearing your wallet on Stellar is long enough to cause some alarm (40+ seconds). The Activity section is usually empty when this happens, so there's no assurance that your asset balances are still pending.

parrot9design commented:
- [ ] When supplying AVAX/bnUSD liquidity, the first transaction finished within 30 seconds, but it wouldn’t let me send the second asset until 1 minute had elapsed.

![Supply liquidity](https://github.com/user-attachments/assets/d50a5ec0-79c0-4b31-9f1d-958a68503622)


- [ ] I now encounter a `failed signing` message in Hana when I try to supply liquidity for SUI/bnUSD (or make any Sui transactions, for that matter), which persists after quitting the browser.

https://github.com/user-attachments/assets/e90a7b53-cc43-4fca-9b37-fab2486d05de

In case it helps, here's the transaction data used in Hana:
```
{
  "version": 2,
  "sender": null,
  "expiration": null,
  "gasData": {
    "budget": null,
    "price": null,
    "owner": null,
    "payment": null
  },
  "inputs": [
    {
      "Pure": {
        "bytes": "JyuBvwAAAAA="
      }
    },
    {
      "Pure": {
        "bytes": "gNHwCAAAAAA="
      }
    },
    {
      "UnresolvedObject": {
        "objectId": "0x25c200a947fd16903d9daea8e4c4b96468cf08d002394b7f1933b636e0a0d500"
      }
    },
    {
      "UnresolvedObject": {
        "objectId": "0xe9ae3e2d32cdf659ad5db4219b1086cc0b375da5c4f8859c872148895a2eace2"
      }
    },
    {
      "UnresolvedObject": {
        "objectId": "0x1bbf52529d14124738fac0abc1386670b7927b6d68cab7f9bd998a0c0b274042"
      }
    },
    {
      "Pure": {
        "bytes": "ATMweDEuaWNvbi9jeGEwYWYzMTY1YzA4MzE4ZTk4OGNiMzA5OTNiMzA0ODMzNWI5NGFmNmM="
      }
    },
    {
      "Pure": {
        "bytes": "AXN7Im1ldGhvZCI6Il9kZXBvc2l0IiwicGFyYW1zIjp7ImFkZHJlc3MiOiJzdWkvMHg1Y2I3MjQxMzFiYTM4OWIwMTRlMTc3M2JjMTRiNDk4ZWVkYmUxZWVkZmU2OGFiOTA5YzEyMGVjYTEyNzk1MzQ1In19"
      }
    }
  ],
  "commands": [
    {
      "SplitCoins": {
        "coin": {
          "GasCoin": true
        },
        "amounts": [
          {
            "Input": 0
          },
          {
            "Input": 1
          }
        ]
      }
    },
    {
      "MoveCall": {
        "package": "0xede387fa2f3789f2e64d46744741d317b21f3022488d8f8ef850b3855ae37919",
        "module": "asset_manager",
        "function": "deposit",
        "typeArguments": [
          "0x2::sui::SUI"
        ],
        "arguments": [
          {
            "Input": 2
          },
          {
            "Input": 3
          },
          {
            "Input": 4
          },
          {
            "NestedResult": [
              0,
              1
            ]
          },
          {
            "NestedResult": [
              0,
              0
            ]
          },
          {
            "Input": 5
          },
          {
            "Input": 6
          }
        ]
      }
    }
  ]
}
```
